src/mon/ConfigKeyService.cc

```diff
@@ -34,61 +34,145 @@ static ostream& _prefix(std::ostream *_dout, const Monitor *mon,
 }
 
 const string ConfigKeyService::STORE_PREFIX = "mon_config_key";
+const string ConfigKeyService::META_PREFIX = "mon_config_key_meta";
+const string ConfigKeyService::META_KEY = "meta";
 
 void ConfigKeyService::refresh()
 {
-  dout(10) << __func__ << dendl;
-  config_keys.clear();
-
-  KeyValueDB::Iterator iter =
-    mon->store->get_iterator(STORE_PREFIX);
-  while (iter->valid()) {
-    config_keys[iter->key()] = iter->value();
-    iter->next();
+  version_t had_version = version;
+  bufferlist bl;
+  mon->store->get(META_PREFIX, META_KEY, bl);
+  list<string> changed_keys;
+  uint32_t total_keys = 0;
+  if (bl.length()) {
+    auto p = bl.begin();
+    ::decode(version, p);
```

Comment:
I'd much have a version controlled structure being encoded/decoded. Encoding and decoding directly seem like it's going to blow in our faces at some point in the future.